### ISSUE TITLE
Cost reduction don't work for opponent's contolled cards

### ISSUE THREAD

**Issue by JayDi85** [magefree/mage#6074]
Affinity bug example #5931 (comments 30e8fec8df9c58e179c28ed4db32a814809ab798)

If you play card from opponent hand then you've got spell ability on the stack under your control, but original card will be under opponent control. That's can cause wrong calc like permanent count (you will call it with wrong controller).

Search potential problem code by `Ability abilityToModify)`:
![shot_191206_223835](https://user-images.githubusercontent.com/8344157/70353286-82f4ff80-1886-11ea-93a6-b5899225dc09.png)

There is different controllerId:
* `source.getControllerId()` -- always return opponent id (if you play it from opponent hand);
* `abilityToModify.getControllerId()` -- return real controller of spell ability.
![shot_191206_223128](https://user-images.githubusercontent.com/8344157/70346816-9bf6b400-1878-11ea-8aac-ca23325e5b5a.png)

Quick fix: get real controller from `abilityToModify.getControllerId()`... It's will be fine with cost reduction effects for yourself (applies on spell ability on the stack), but need more research for other cards/effects.

**Comment by jeffwadsworth** [comment]
https://github.com/magefree/mage/commit/57cb023f78c1e68a990ab66760886b38388eee33
Does this not work for you?

**Comment by JayDi85** [comment]
Yes, it's works fine as `getActivePermanents` or `count` calls. Problem was in wrong controller. Tested it for Tezzeret with Sen Triplets

```
[test control]
battlefield:Human:Sen Triplets:1
battlefield:Human:Tezzeret, Master of the Bridge:1
hand:Computer:Adarkar Valkyrie:1
hand:Human:Adarkar Valkyrie:1
hand:Computer:Aerie Ouphes:1
battlefield:Computer:Aesthir Glider:2
battlefield:Human:Aesthir Glider:4
battlefield:Human:Plains:4
```
